src/core_editor/line_buffer.rs

```diff
@@ -541,26 +541,67 @@ mod test {
         assert_eq!(expected_line_buffer, line_buffer);
     }
 
-    #[test]
-    #[ignore] // Note: Not sure if this is the intended behaviour
-    fn uppercase_word_works_when_one_last_index() {
-        let mut line_buffer = buffer_with("This is a test");
-        line_buffer.uppercase_word();
+    #[rstest]
+    // FIXME: capitalize_char move insertion point to the end
+    #[case("This is a test", (0, 13), "This is a tesT", (0, 14))]
+    // FIXME: capitalize_char move insertion point to the end
+    #[case("This is a test", (0, 10), "This is a Test", (0, 14))]
```

Comment:
Should the cursor position change when doing capitalization?

Reply:
I was thinking it should, like `~` in vim

Reply:
In vim `~` capitalizes then move left 1 step, whereas this function capitalizes and then move to the **end**. 

Also, I am not so sure what approach should the editor take like we can have one where we do two things 
- capitalize and then move left 
- (or) only capitalize and leave the moving left to the caller.

Thoughts on why we should pick one over the other?